### PR TITLE
First batch of unit tests

### PR DESCRIPTION
**Pull request by peskaf**
Added unit tests for most functions and classes in `utils` together with updated `BCE with logits` loss tests.

Corrected some doc strings that did not match the code or had typos.

### CODE REVIEW

**Review by kozlov721 — APPROVED** [review]
LGTM
